src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -165,6 +182,48 @@ internal Batch<Metric> Collect()
             }
         }
 
+        /// <summary>
+        /// Called by <c>ForceFlush</c>. This function should block the current
+        /// thread until flush completed or timed out.
+        /// </summary>
+        /// <param name="timeoutMilliseconds">
+        /// The number of milliseconds to wait, or <c>Timeout.Infinite</c> to
+        /// wait indefinitely.
+        /// </param>
+        /// <returns>
+        /// Returns <c>true</c> when flush succeeded; otherwise, <c>false</c>.
+        /// </returns>
+        /// <remarks>
+        /// This function is called synchronously on the thread which made the
+        /// first call to <c>ForceFlush</c>. This function should not throw
+        /// exceptions.
+        /// </remarks>
+        internal bool OnForceFlush(int timeoutMilliseconds)
```

Comment:
we need some public method like MeterProviderExtensions.ForceFlush, which in-turn calls this..

Reply:
Yes, that's too much for this PR.
`MeterProviderSdk` is still a public class, need to make it internal.